dashboard/test/models/pd/application/teacher_application_test.rb

```diff
@@ -109,6 +109,17 @@ class TeacherApplicationTest < ActiveSupport::TestCase
       assert_equal Pd::Application::TeacherApplication::REVIEWING_INCOMPLETE, teacher_application.meets_criteria
     end
 
+    test 'date_applied finds date of first unreviewed status' do
+      tomorrow = Date.tomorrow.to_time
+      application = create :pd_teacher_application, status: 'incomplete'
+      assert_nil application.date_applied
+
+      Timecop.freeze(tomorrow) do
+        application.update!(status: 'unreviewed')
+        assert_equal tomorrow, application.date_applied.to_time
+      end
+    end
```

Comment:
nice test! I would suggest adding a test case for reopening / resubmitting.

Reply:
Good call! I modified this test for re-opening and re-submitting. Let me know if you think they should be separated out